sentinel-core/src/main/java/com/alibaba/csp/sentinel/CtEntry.java

```diff
@@ -75,8 +79,9 @@ protected void exitForContext(Context context, int count, Object... args) throws
                 if (parent != null) {
                     ((CtEntry)parent).child = null;
                 }
-                if (parent == null) {
-                    // Auto-created entry indicates immediate exit.
+                if (parent == null && !(context instanceof NullContext)) {
```

Comment:
Check `NullContext` at the beginning of this method, if current context is `NullContext`, this method should return at once.